File: backend/node/models/problemModel.js
const { getDB } = require('../db/mongodb');
const { ObjectId } = require('mongodb');

const getSolvedProblems = async (userId) => {
  const db = getDB();
  if (!db) return [];
  
  const problems = await db.collection('solved_problems')
    .find({ user_id: new ObjectId(userId) })
    .toArray();
  return problems;
};

const solveProblem = async (userId, problemId, attempts, timeTaken) => {
  const db = getDB();
  if (!db) return { problemId, attempts, timeTaken };
  
  const existing = await db.collection('solved_problems').findOne({
    user_id: new ObjectId(userId),
    problem_id: problemId
  });
  
  if (!existing) {
    await db.collection('solved_problems').insertOne({
      user_id: new ObjectId(userId),
      problem_id: problemId,
      attempts,
      time_taken: timeTaken,
      solved_at: new Date()
    });
  }
  
  return { problemId, attempts, timeTaken };
};

module.exports = { getSolvedProblems, solveProblem };

File: backend/python/models/problem_model.py
from db.mongodb import get_db
from bson import ObjectId
from datetime import datetime

def solve_problem(user_id, problem_id, attempts, time_taken):
    db = get_db()
    solved = {
        'user_id': ObjectId(user_id),
        'problem_id': problem_id,
        'attempts': attempts,
        'time_taken': time_taken,
        'solved_at': datetime.utcnow()
    }
    result = db.solved_problems.insert_one(solved)
    solved['_id'] = result.inserted_id
    return solved

def get_solved_problems(user_id):
    db = get_db()
    problems = list(db.solved_problems.find({'user_id': ObjectId(user_id)}))
    return problems
